### PR TITLE
ACS Calling Release 2.0.0-beta.1

### PR DESCRIPTION
**Pull request by raosanat**

### CODE REVIEW

**Review by tjprescott — CHANGES_REQUESTED** [review]
If this is a breaking change, the version should be 2.0.0-beta.1. This would also need archboard approval.

File: sdk/communication/AzureCommunicationCalling/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,27 @@
...
+### Breaking changes
+- `didChangeRecordingState` and `didChangeTranscriptionState` are moved out of `CallDelegate` and into `RecordingFeatureDelegate` and `TranscriptionFeatureDelegate`.
+
+### Bug fixes
+- [iOS] Not triggering calldidChangeState if using createCallAgentWithCallKitOption. And not able to accept the call in App killed state https://github.com/Azure/Communication/issues/316.
+- [iOS] Remote participant stream is stretched after hang-up https://github.com/Azure/Communication/issues/311.
+- CallKit bug fixes with sometimes call gets disconnected immediately after picking up the call.
+- Intermittent freezing of UI thread issue is fixed when for e.g. reading property `remoteParticipants` in a `Call`.
+- `Hold` and `Resume` support fixed with CallKit enabled `CallAgent`'s.
+
```
Comment (tjprescott) [inline sdk/communication/AzureCommunicationCalling/CHANGELOG.md:24]: This needs to follow the new guidance on Changelogs here: https://azure.github.io/azure-sdk/policies_releases.html#change-logs

**Review by tjprescott — CHANGES_REQUESTED** [review]

**Comment by tjprescott** [comment]
@JeffreyRichter adding ArchApproved tag

File: sdk/communication/AzureCommunicationCalling/CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,28 @@
...
+            let recordingFeature = self.call!.api(RecordingFeature.self)
+            recordingFeature.delegate = self.callObserver
+```
+
+### Breaking changes
+- `didChangeRecordingState` and `didChangeTranscriptionState` are moved out of `CallDelegate` into `RecordingFeatureDelegate` and into `TranscriptionFeatureDelegate`.
+
+**More documentation on extensions and the breaking change can be found [here](https://docs.microsoft.com/en-us/azure/communication-services/quickstarts/voice-video-calling/calling-client-samples?pivots=platform-ios).**
+
+*NOTE: Version previously released `1.2.0-beta.1` and this version only has the difference in version number and this version will be available in cocoapods.*
```
Comment (tjprescott) [inline sdk/communication/AzureCommunicationCalling/CHANGELOG.md:17]: This note is a little confusing. I'd recommend:
"NOTE: Previously released version `1.2.0-beta.1` is functionally identical to `2.0.0-beta.1`."

File: sdk/communication/AzureCommunicationCalling/CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,28 @@
 # Release History
 
+## 2.0.0-beta.1 (2021-07-14)
+### New features
```
Comment (tjprescott) [inline sdk/communication/AzureCommunicationCalling/CHANGELOG.md:4]: Per updated guidance this should be "Features Added"

File: sdk/communication/AzureCommunicationCalling/CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,28 @@
...
+- `Recording` and `Transcription` features are decoupled from `Call` object and now can only be used via extensions.
+  For e.g. 
+```
+            let recordingFeature = self.call!.api(RecordingFeature.self)
+            recordingFeature.delegate = self.callObserver
+```
+
+### Breaking changes
```
Comment (tjprescott) [inline sdk/communication/AzureCommunicationCalling/CHANGELOG.md:12]: Nit: Breaking Changes

File: sdk/communication/AzureCommunicationCalling/CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,28 @@
...
+- `didChangeRecordingState` and `didChangeTranscriptionState` are moved out of `CallDelegate` into `RecordingFeatureDelegate` and into `TranscriptionFeatureDelegate`.
```
Comment (tjprescott) [inline sdk/communication/AzureCommunicationCalling/CHANGELOG.md:13]: Per updated guidance:
Should be prefaced with `Changed:`

File: sdk/communication/AzureCommunicationCalling/CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,28 @@
...
+
+**More documentation on extensions and the breaking change can be found [here](https://docs.microsoft.com/en-us/azure/communication-services/quickstarts/voice-video-calling/calling-client-samples?pivots=platform-ios).**
+
+*NOTE: Version previously released `1.2.0-beta.1` and this version only has the difference in version number and this version will be available in cocoapods.*
+
+### Bug fixes
```
Comment (tjprescott) [inline sdk/communication/AzureCommunicationCalling/CHANGELOG.md:19]: Per updated guidance:
Should be "Bugs Fixed"

File: sdk/communication/AzureCommunicationCalling/CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,28 @@
...
+- [iOS] Not triggering calldidChangeState if using createCallAgentWithCallKitOption. And not able to accept the call in App killed state https://github.com/Azure/Communication/issues/316.
```
Comment (tjprescott) [inline sdk/communication/AzureCommunicationCalling/CHANGELOG.md:20]: This format should follow the example in the updated guidance:
```
- Issue where AccountName on BlobUriBuilder would not be populated
  for non-IP style Uris ([#8638](https://github.com/Azure/azure-sdk-for-net/issues/8638))
```
We should remove the [iOS] tag and make sure the URL is represented as shown above.

File: sdk/communication/AzureCommunicationCalling/CHANGELOG.md (hunk 5 above)
Comment (raosanat) [reply sdk/communication/AzureCommunicationCalling/CHANGELOG.md:13]: Did not understand ,where Changed should be ?

File: sdk/communication/AzureCommunicationCalling/CHANGELOG.md (hunk 7 above)
Comment (raosanat) [reply sdk/communication/AzureCommunicationCalling/CHANGELOG.md:20]: [iOS] tag comes from the bug title. But i can change the link format

File: sdk/communication/AzureCommunicationCalling/CHANGELOG.md (hunk 5 above)
Comment (tjprescott) [reply sdk/communication/AzureCommunicationCalling/CHANGELOG.md:16]: - Changed: `didChangeRecordingState` and...

Comment (raosanat) [reply sdk/communication/AzureCommunicationCalling/CHANGELOG.md:16]: ok.

File: sdk/communication/AzureCommunicationCalling/CHANGELOG.md (hunk 7 above)
Comment (tjprescott) [reply sdk/communication/AzureCommunicationCalling/CHANGELOG.md]: The changelog should not just replicate the bug title, as that is not always a logical explanation of the bug. This should be a concise description of the issue.

File: sdk/communication/AzureCommunicationCalling/CHANGELOG.md (hunk 2 above)
Comment (tjprescott) [reply sdk/communication/AzureCommunicationCalling/CHANGELOG.md]: This was moved but still confusing.

**Review by tjprescott — APPROVED** [review]

**Pull request merged**